xbmc/messaging/helpers/DialogHelper.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+/*
+*      Copyright (C) 2005-2015 Team Kodi
+*      http://kodi.tv
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with Kodi; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include "utils/Variant.h"
+
+#include <string>
+
+namespace KODI
+{
+namespace MESSAGING
+{
+namespace HELPERS
+{
+static const std::string DIALOG_HEADING{"heading"};
+static const std::string DIALOG_TEXT{"text"};
+static const std::string DIALOG_LINE0{"line0"};
+static const std::string DIALOG_LINE1{"line1"};
+static const std::string DIALOG_LINE2{"line2"};
+static const std::string DIALOG_YESLABEL{"yeslabel"};
+static const std::string DIALOG_NOLABEL{"nolabel"};
+static const std::string DIALOG_AUTOCLOSE{"autoclose"};
+
+/*!
+  \brief This is a helper method to send a threadmessage to open a Yes/No dialog box
+
+  \param[in]  heading           The text to display as the dialog box header
+  \param[in]  text              The text to display in the dialog body
+  \param[in]  noLabel           The text to display on the No button
+                                defaults to No
+  \param[in]  yesLabel          The text to display on the Yes button
+                                defaults to Yes
+  \param[in]  autoCloseTimeout  The time before the dialog closes
+                                defaults to 0 show indefinitely
+  \return -1 on cancelled, 0 on no and 1 on yes
+  \sa ShowYesNoDialogLines
+  \sa CGUIDialogYesNo::ShowAndGetInput
+*/
+int ShowYesNoDialogText(CVariant heading, CVariant text, CVariant noLabel = "", CVariant yesLabel = "", uint32_t autoCloseTimeout = 0);
```

Comment:
There's nothing preventing using overloads here but I chose this naming specifically to avoid the overload hell we currently have where it's really easy to call the wrong showandgetinput